app/models/bgs_dependents/marriage.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module BGSDependents
+  class Marriage < Base
+    attribute :ssn, String
+    attribute :first, String
+    attribute :middle, String
+    attribute :last, String
+    attribute :suffix, String
+    attribute :vet_ind, String
+    attribute :birth_date, String
+    attribute :alt_address, String
+    attribute :va_file_number, String
+    attribute :lives_with_vet, String
+    attribute :ever_married_ind, String
+    attribute :martl_status_type_cd, String
+
+    def initialize(dependents_application)
+      @dependents_application = dependents_application
+      @spouse_information = @dependents_application['spouse_information']
+
+      self.attributes = spouse_attributes
+    end
+
+    def format_info
+      attributes.with_indifferent_access
+    end
+
+    def alt_address
+      @dependents_application.dig('does_live_with_spouse', 'address')
+    end
+
+    def address(marriage_info)
+      dependent_address(
+        @dependents_application,
+        marriage_info['lives_with_vet'],
+        marriage_info['alt_address']
+      )
+    end
+
+    private
+
+    def spouse_attributes
+      marriage_info = {
+        'ssn': @spouse_information['ssn'],
+        'birth_date': @spouse_information['birth_date'],
+        'ever_married_ind': 'Y',
+        'martl_status_type_cd': marital_status,
+        'vet_ind': spouse_is_veteran,
+        'lives_with_vet': lives_with_vet,
+        'alt_address': alt_address
+      }.merge(@spouse_information['full_name']).with_indifferent_access
+
+      marriage_info.merge!({ 'va_file_number': @spouse_information['va_file_number'] }) if spouse_is_veteran == 'Y'
+
+      marriage_info
+    end
+
+    def lives_with_vet
+      @dependents_application.dig('does_live_with_spouse', 'spouse_does_live_with_veteran')
```

Comment:
Curious about your usage of `dig` rather than referencing the hash directly? Is `@dependents_application['does_live_with_spouse']['spouse_does_live_with_veteran']` not guaranteed to exist?

Reply:
I am paranoid of calling a method on `nil`, after researching and speaking w/ our FE team. I feel confident in `does_live_with_spouse => spouse_does_live_with_veteran ` being present.

Reply:
> I am just paranoid of calling a method on nil

It's probably better to fail fast than to keep marching along like things are fine. In that vein, I'll go broken record on you, and recommend some active model validations, so that once initialization is complete, you can simply call `raise 'we didnt get all the stuff we were expecting' unless self.valid?`